.github/workflows/main.yml

```diff
@@ -22,22 +45,16 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          cache: pip
 
       - name: Install project dependencies
         run: |
-          pip3 install --upgrade pip
-          pip3 install codecov tox tox-gh-actions
+          pip install --upgrade pip
+          pip install codecov '.[tests]' '${{ matrix.urllib3-requirement }}'
+          pip check
 
-      - name: Run online tests with tox
-        run: tox -- -m online
-
-      - name: Run offline tests with tox with no access to the Internet
-        run: |
-          # We're using unshare to take Internet access
-          # away from tox so that we'll notice whenever some new test
-          # is missing @pytest.mark.online decoration in the future
-          unshare --map-root-user --net -- \
-              sh -c 'ip link set lo up; tox -- -m "not online"'
+      - name: Run tests
+        run: ./runtests.sh --cov=./vcr --cov-branch --cov-report=xml --cov-append
```

Comment:
@jairhenrique could you bring back the split between `-m online` and `-m 'not online'` please? Where loosing something of value here. Thanks!